beacon-chain/db/kv/state.go

```diff
@@ -97,6 +100,39 @@ func (k *Store) SaveState(ctx context.Context, state *pb.BeaconState, blockRoot
 	})
 }
 
+// DeleteState by block root.
+func (k *Store) DeleteState(ctx context.Context, blockRoot [32]byte) error {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.DeleteState")
+	defer span.End()
+	return k.db.Batch(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(stateBucket)
+		return bkt.Delete(blockRoot[:])
+	})
+}
+
+// DeleteStates by block roots.
+func (k *Store) DeleteStates(ctx context.Context, blockRoots [][32]byte) error {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.DeleteStates")
+	defer span.End()
+	var wg sync.WaitGroup
+	errs := make([]string, 0)
+	wg.Add(len(blockRoots))
+	for _, r := range blockRoots {
+		go func(w *sync.WaitGroup, root [32]byte) {
+			defer w.Done()
+			if err := k.DeleteState(ctx, root); err != nil {
+				errs = append(errs, err.Error())
```

Comment:
maybe can add a test for this code path?

Reply:
yea ill add one in the following PR

Reply:
although it's hard to simulate that `err`

Reply:
you are right `Returns an error if the bucket was created from a read-only transaction`

Reply:
no need i guess